cirq-core/cirq/experiments/parallel_readout_calibration.py

```diff
@@ -0,0 +1,136 @@
+        repetitions: The number of measurement repetitions to perform for
+            each trial.
+        trials: The number of bitstrings to prepare.
+        trials_per_batch:  If provided, split the experiment into batches
+            with this number of trials in each batch.
+        bit_strings: A list of ints that specifies the bit strings for each
+            qubit.  There should be exactly one int per qubit.  Bits
+            represent whether the qubit is |0〉or |1〉when initialized.
+            Trials are done from least significant bit to most significant bit.
+            If not provided, the function will generate random bit strings
+            for you.
+
+    Returns:
+        A ReadoutExperimentResult storing the readout error
+        probabilities as well as the number of repetitions used to estimate
+        the probabilities. Also stores a timestamp indicating the time when
+        data was finished being collected from the sampler.  Note that,
+        if there did not exist a trial where a given qubit was set to |0〉,
+        the zero-state error will be set to `nan` (not a number).  Likewise
+        for qubits with no |1〉trial and one-state error.
+    """
+    qubits = list(qubits)
+
+    if trials <= 0:
+        raise ValueError("Must provide non-zero trials for readout calibration.")
+    if repetitions <= 0:
+        raise ValueError("Must provide non-zero repetition for readout calibration.")
+    if bit_strings is None:
+        bit_strings = [random.getrandbits(trials) for _ in qubits]
+    if len(bit_strings) != len(qubits):
+        raise ValueError(
+            f'If providing bit_strings, # of bit strings ({len(bit_strings)}) '
+            f'must equal # of qubits ({len(qubits)})'
+        )
+
+    all_circuits = []
+    all_sweeps: List[study.Sweepable] = []
+    if trials_per_batch is not None:
+        if trials_per_batch <= 0:
+            raise ValueError("Must provide non-zero trials_per_batch for readout calibration.")
+        num_batchs = trials // trials_per_batch
+        if trials % trials_per_batch > 0:
+            num_batchs += 1
+    else:
+        num_batchs = 1
+        trials_per_batch = trials
+    for batch in range(num_batchs):
+        circuit = circuits.Circuit()
+        single_sweeps = []
+        for idx, q in enumerate(qubits):
+            sym_val = f'bit_{idx}'
+            current_trial = bit_strings[idx]
+            trial_range = range(batch * trials_per_batch, (batch + 1) * trials_per_batch)
+            circuit.append(ops.X(q) ** sympy.Symbol(sym_val))
+            single_sweeps.append(
+                study.Points(
+                    key=sym_val, points=[(current_trial >> bit) & 1 for bit in trial_range]
+                )
+            )
+
+        circuit.append(ops.measure_each(*qubits, key_func=repr))
+        total_sweeps = study.Zip(*single_sweeps)
+        all_circuits.append(circuit)
+        all_sweeps.append(total_sweeps)
```

Comment:
This is more of a stylistic choice here so feel free to ignore, what do you think of seperating things out a little bit to something like:

```python3
flip_symbols = sympy.symbols(f'flip_0:{len(qubits)}')
flip_circuit = cirq.Circuit([cirq.X(q) for q in qubits], [ops.measure_each(*qubits, key_func=repr)])
circuits = [flip_circuit] * num_batches

for batch in range(num_batches):
  # single_sweeps = []
  # for i, current_symbol in enumerate(flip_symbols):
  #   current_sweep = study.Points(key=current_symbol, points=flip_values[batch][i])
  #   single_sweeps.append(ref_sweep)
  single_sweeps = [study.Points(key=symb, points=flip_values[batch][i]) for i, symb in enumerate(flip_symbols)]
  total_sweeps = study.Zip(*single_sweeps)
  all_sweeps.append(total_sweeps)
```
Where `flip_values` is a 3d numpy boolean array built from `bit_strings`, who's first dimension indexes batch, second dimension indexes qubit and third dimension indexes flip value (T/F) ?

Reply:
Basically did this, but kept it as a 2d array without the batch dimension.